packages/hoprd/src/identity.ts

```diff
@@ -67,27 +81,40 @@ async function storeIdentity(path: string, id: Uint8Array) {
   fs.writeFileSync(resolve(path), id)
 }
 
-async function createIdentity(idPath: string, password: string, useWeakCrypto = false) {
-  const peerId = await PeerId.create({ keyType: 'secp256k1' })
+async function createIdentity(idPath: string, password: string, useWeakCrypto = false, privateKey?: Uint8Array) {
+  const peerId = privateKey ? privKeyToPeerId(privateKey) : await PeerId.create({ keyType: 'secp256k1' })
   const serializedKeyPair = await serializeKeyPair(peerId, password, useWeakCrypto)
   await storeIdentity(idPath, serializedKeyPair)
   return peerId
 }
 
 export async function getIdentity(options: IdentityOptions): Promise<PeerId> {
+  let privateKey: Uint8Array | undefined
+  if (options.privateKey) {
+    privateKey = Uint8Array.from(
+      Buffer.from(options.privateKey.substr(0, 2) === '0x' ? options.privateKey.substr(2) : options.privateKey, 'hex')
```

Comment:
Should be handled by `@hoprnet/hopr-utils->stringToU8a` - also requires Buffer to be present in our environment

Reply:
Will check `stringToU8a`, should be straightforward to replace it then